src/Data/ProgrammesDb/EntityRepository/CollapsedBroadcastRepository.php

```diff
@@ -167,6 +167,24 @@ public function findByCategoryAncestryAndStartAtDateRange(
         return $this->resolveProgrammeParents($result);
     }
 
+    public function findByProgramme(
+        array $ancestry,
+        bool $isWebcastOnly,
+        ?int $limit,
+        int $offset
+    ): array {
+        $qb = $this->createCollapsedBroadcastsOfProgrammeQueryBuilder($ancestry, $isWebcastOnly)
```

Comment:
If you're going to use this just for showing the broadcasts in the episode page, you probably don't need all of the joins that this method adds. Can you make this query slimmer?

Reply:
Yeah, I don't think createCollapsedBroadcastsOfProgrammeQueryBuilder is right here. You are looking for broadcasts of a single episode, that will give you broadcasts of an episode and of all of its children. Not that the episode will have children, but that's a much slower query. We just need to ascertain that the collapsed broadcast has a programme item ID the same as our programme. 

I would probably add the same joins as that thing does though, chances are some of our presenters will be expecting them and relational databases are exceptionally fast at doing joins if you're not trying to filter by both sides of them, which we aren't.